internal/resource/config/datastore/ping_one_ldap_gateway_data_store.go

```diff
@@ -0,0 +1,264 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/objectvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/objectdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/resourcelink"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+var (
+	pingOneLdapGatewayDataStoreAttrType = map[string]attr.Type{
+		"ping_one_connection_ref":  basetypes.ObjectType{AttrTypes: resourcelink.AttrType()},
+		"ldap_type":                basetypes.StringType{},
+		"ping_one_ldap_gateway_id": basetypes.StringType{},
+		"use_ssl":                  basetypes.BoolType{},
+		"name":                     basetypes.StringType{},
+		"binary_attributes":        basetypes.SetType{ElemType: basetypes.StringType{}},
+		"type":                     basetypes.StringType{},
+		"ping_one_environment_id":  basetypes.StringType{},
+	}
+	pingOneLdapGatewayDataStoreEmptyStateObj = types.ObjectNull(pingOneLdapGatewayDataStoreAttrType)
+)
+
+func toSchemaPingOneLdapGatewayDataStore() schema.SingleNestedAttribute {
+	pingOneLdapGatewayDataStoreSchema := schema.SingleNestedAttribute{}
+	pingOneLdapGatewayDataStoreSchema.Description = "A PingOne LDAP Gateway data store."
+	pingOneLdapGatewayDataStoreSchema.Default = objectdefault.StaticValue(types.ObjectNull(pingOneLdapGatewayDataStoreAttrType))
+	pingOneLdapGatewayDataStoreSchema.Computed = true
+	pingOneLdapGatewayDataStoreSchema.Optional = true
+	pingOneLdapGatewayDataStoreSchema.Attributes = map[string]schema.Attribute{
+		"type": schema.StringAttribute{
+			Description: "The data store type.",
+			Computed:    true,
+			Optional:    false,
+			Default:     stringdefault.StaticString("PING_ONE_LDAP_GATEWAY"),
+		},
+		"name": schema.StringAttribute{
+			Description: "The data store name with a unique value across all data sources. Omitting this attribute will set the value to a combination of the hostname(s) and the principal.",
+			Computed:    true,
+			Optional:    true,
```

Comment:
If these defaults depend on other TF attributes, we could use a ModifyPlan to set them